### PR TITLE
Added exceptions for Exception and ValueError

### PR DESCRIPTION
**Pull request by kompotkot**
Reporter take tags additional argument.

### CODE REVIEW

File: python/humbug/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from .report import Reporter
+
+
+class ExceptionWithReporting(Exception):
+    def __init__(self, bugout_reporter: Reporter, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        bugout_reporter.error_report(self, publish=True)
+
+
+class ValueErrorWithReporting(ValueError):
```
Comment (zomglings) [inline python/humbug/exceptions.py:13]: I don't think we should create a different exception class for each of the Python native exception classes. It's best to just define `ExceptionWithReporting` and let users inherit from it as they want.

In my experience, most people just define their own custom exception hierarchies starting from `Exception` in the first place.

File: python/humbug/report.py [hunk 2]
```diff
@@ -44,6 +45,7 @@ def __init__(
         self.bugout_token = bugout_token
         self.bugout_journal_id = bugout_journal_id
         self.timeout_seconds = timeout_seconds
+        self.additional_tags = tags
```
Comment (zomglings) [inline python/humbug/report.py:48]: Do:
```python
self.additional_tags = []
if tags is not None:
    self.additional_tags = tags
```

File: python/humbug/report.py [hunk 3]
```diff
@@ -33,6 +33,7 @@ def __init__(
         bugout_token: Optional[str] = None,
         bugout_journal_id: Optional[str] = None,
         timeout_seconds: int = 10,
+        tags: Optional[List[str]] = None
```
Comment (zomglings) [inline python/humbug/report.py:36]: Can you also add the following two parameters:
1. `client_id: str`,
2. `session_id: str`.

File: python/humbug/report.py [hunk 4]
```diff
@@ -44,6 +45,7 @@ def __init__(
         self.bugout_token = bugout_token
         self.bugout_journal_id = bugout_journal_id
         self.timeout_seconds = timeout_seconds
+        self.additional_tags = tags
 
     def system_tags(self) -> List[str]:
         tags = [
```
Comment (zomglings) [inline python/humbug/report.py:51]: I don't want to store `client_id:{self.client_id}` and `session_id:{self.session_id}` in `self.additional_tags`. We should populate them into `tags` array separately.

Add lines like:
```python
def session_tags(self):
    return ["client_id:{}".format(self.client_id), "session_id:{}".format(self.session_id)]
```

```
report.tags += self.session_tags()
```

**Review by zomglings — CHANGES_REQUESTED** [review]
@kompotkot : Left some comments. Also, needs tests.

File: python/humbug/report.py (hunk 2 above)
Comment (kompotkot) [reply python/humbug/report.py:48]: Done

File: python/humbug/report.py (hunk 3 above)
Comment (kompotkot) [reply python/humbug/report.py:36]: Was added in v0.1.3

File: python/humbug/exceptions.py (hunk 1 above)
Comment (kompotkot) [reply python/humbug/exceptions.py:13]: Saved only ExceptionWithReporting

File: python/humbug/report.py (hunk 4 above)
Comment (kompotkot) [reply python/humbug/report.py:51]: As I understand it is not actual any more, after v0.1.3 was released

**Pull request closed without merging**